doc/examples/transform/plot_tps_deformation.py

```diff
@@ -0,0 +1,147 @@
+r"""
+==========================================
+Interpolate images with thin-plate splines
```

Comment:
This might get into semantics but I think interpolation just means something like finding new values based on existing often neighboring data. That's not necessarily reconstruction only but also deformation / remapping, / relocating. I don't find "deform" quite fitting if TPS can also be used to shift or rotate an image. So I'm okay with "interpolate" being the generalized term. Whether this gallery example demonstrates all use cases of "interpolation" is another thing...

Reply:
The example demonstrate interpolating a 2D image over a set `src` and `dst` points. 
@mkcor I update this section based on  https://github.com/scikit-image/scikit-image/pull/7040#discussion_r1509945863

Reply:
Following the convention of writing titles at the imperative mood, I'd suggest either "Warp images using thin-plate splines" or "Use thin-plate splines to warp images" (or "Use thin-plate splines for image warping?").